### PR TITLE
fix(ecs-gestalt): Migrate Components to gestalt's Components.

### PR DESCRIPTION
**Pull request by DarkWeird**
Ref: MovingBlocks/Terasology#4753

### CODE REVIEW

File: src/main/java/org/terasology/computer/display/component/DisplayComponent.java [hunk 1]
```diff
@@ -62,4 +51,12 @@ public void setMode(String mode) {
     public void setData(List<String> data) {
         this.data = data;
     }
+
+    @Override
+    public void copy(DisplayComponent other) {
+        this.monitorSize = other.monitorSize;
+        this.front = other.front;
+        this.mode = other.mode;
+        this.data = Lists.newArrayList(data);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/computer/display/component/DisplayComponent.java]: If you change the constructor to use `addAll` , why not `.clear()` and `.addAll(other.data)` here? 

Also, there's a bug in there as it should at least be 
```suggestion
        this.data = Lists.newArrayList(other.data);
```

File: src/main/java/org/terasology/computer/display/component/DisplayDataHolderComponent.java [hunk 2]
```diff
@@ -52,4 +41,12 @@ public String getMode() {
     public List<String> getData() {
         return data;
     }
+
+    @Override
+    public void copy(DisplayDataHolderComponent other) {
+        this.monitorSize.set(other.monitorSize);
+        this.front = other.front;
+        this.mode = other.mode;
+        this.data = Lists.newArrayList(other.data);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/computer/display/component/DisplayDataHolderComponent.java]: Same question as above.

**Review by skaldarnar — CHANGES_REQUESTED** [review]

File: src/main/java/org/terasology/computer/display/component/DisplayComponent.java (hunk 1 above)
Comment (jdrueckert) [reply src/main/java/org/terasology/computer/display/component/DisplayComponent.java]: like so?

File: src/main/java/org/terasology/computer/display/component/DisplayDataHolderComponent.java (hunk 2 above)
Comment (jdrueckert) [reply src/main/java/org/terasology/computer/display/component/DisplayDataHolderComponent.java]: like so?

**Review by jdrueckert — DISMISSED** [review]

File: src/main/java/org/terasology/computer/display/component/DisplayComponent.java [hunk 3]
```diff
@@ -62,4 +51,13 @@ public void setMode(String mode) {
     public void setData(List<String> data) {
         this.data = data;
     }
+
+    @Override
+    public void copyFrom(DisplayComponent other) {
+        this.monitorSize = other.monitorSize;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/computer/display/component/DisplayComponent.java]: As we always initialize the size we can reuse our own instance and just set the value.
```suggestion
        this.monitorSize.set(other.monitorSize);
```

**Review by skaldarnar — APPROVED** [review]

**Pull request merged**